src/psyclone/psyGen.py

```diff
@@ -6069,6 +6315,44 @@ def __init__(self, operator, parent=None):
         self._text_name = "NaryOperation"
 
 
+    @staticmethod
+    def create(oper, children):
+        '''Create an NaryOperator instance given an operator and a list of
+        Node instances.
+
+        :param operator: the operator used in the operation.
+        :type operator: :py:class:`psyclone.psyGen.NaryOperation.Operator`
+        :param children: a list of PSyIR nodes that the operator \
+            operates on.
+        :type children: list of :py:class:`psyclone.psyGen.Node`
+
+        :returns: an NaryOperator instance.
+        :rtype: :py:class:`psyclone.psyGen.NaryOperator`
+
+        '''
+        if not isinstance(oper, NaryOperation.Operator):
+            raise GenerationError(
+                "oper argument to class NaryOperation method create "
+                "should be a PSyIR NaryOperation Operator but found '{0}'."
+                "".format(type(oper).__name__))
+        if not isinstance(children, list):
+            raise GenerationError(
+                "children argument to class NaryOperation method "
+                "create should be a list but found '{0}'."
+                "".format(type(children).__name__))
+        for child in children:
+            if not isinstance(child, Node):
+                raise GenerationError(
+                    "child of children argument to class NaryOperation method "
+                    "create should be a PSyIR Node but found '{0}'."
+                    "".format(type(child).__name__))
+        nary_op = NaryOperation(oper)
+        for child in children:
+            child.parent=nary_op
+        nary_op.children = children
+        return nary_op
+
+
 class Array(Reference):
```

Comment:
Does the Array class needs a create method as well? It is not a leaf node

Reply:
Yes, and KernelSchedule too I presume.